src/common/redis_module/ray_redis_module.cc

```diff
@@ -431,6 +470,68 @@ int TableLookup_RedisCommand(RedisModuleCtx *ctx,
   return REDISMODULE_OK;
 }
 
+bool is_nil(const std::string &data) {
+  CHECK(data.size() == 20);
+  const uint8_t *d = reinterpret_cast<const uint8_t *>(data.data());
+  for (int i = 0; i < 20; ++i) {
+    if (d[i] != 255) {
+      return false;
+    }
+  }
+  return true;
+}
+
+// This is a temporary redis command that will be removed once
+// the GCS uses https://github.com/pcmoritz/credis.
+// Be careful, this only supports Task Table payloads.
+int TableTestAndUpdate_RedisCommand(RedisModuleCtx *ctx,
+                                    RedisModuleString **argv,
+                                    int argc) {
+  if (argc != 3) {
+    return RedisModule_WrongArity(ctx);
+  }
+  RedisModuleString *id = argv[1];
+  RedisModuleString *update_data = argv[2];
+
+  RedisModuleKey *key =
+      OpenPrefixedKey(ctx, "T:", id, REDISMODULE_READ | REDISMODULE_WRITE);
+
+  size_t value_len = 0;
+  char *value_buf = RedisModule_StringDMA(key, &value_len, REDISMODULE_READ);
+
+  size_t update_len = 0;
+  const char *update_buf = RedisModule_StringPtrLen(update_data, &update_len);
+
+  auto data = flatbuffers::GetMutableRoot<TaskTableData>(
+      reinterpret_cast<void *>(value_buf));
+
+  auto update = flatbuffers::GetRoot<TaskTableTestAndUpdate>(update_buf);
+
+  bool do_update = data->scheduling_state() & update->test_state_bitmask();
+
+  if (!is_nil(update->test_scheduler_id()->str())) {
+    do_update =
+        do_update &&
+        update->test_scheduler_id()->str() == data->scheduler_id()->str();
+  }
+
+  if (do_update) {
+    data->mutate_scheduling_state(update->update_state());
```

Comment:
We probably want to `CHECK` the result of any call to `mutate_`... as it may not actually happen (e.g., if the original value starts off with the default value

Reply:
yeah, that's what the ForceDefaults call is for, but checking is a good idea anyways yeah